packages/dataviews/src/components/dataform/stories/index.story.tsx

```diff
@@ -74,11 +103,47 @@ const fields = [
 		elements: [
 			{ value: 'draft', label: 'Draft' },
 			{ value: 'published', label: 'Published' },
+			{ value: 'private', label: 'Private' },
 		],
 	},
-];
+	{
+		id: 'password',
+		label: 'Password',
+		type: 'text' as const,
+		isVisible: ( item: SamplePost ) => {
+			return item.status !== 'private';
+		},
+		dependencies: [ 'status' ],
```

Comment:
I'd remove `dependencies` from this PR. It's a nice idea but, if introduced, it should be connected to the rendering of the field. Right now it only controls the execution of `isVisible`, which seems less impactful than the field rendering.

Reply:
Hmm true, I am ok removing this for now, but would be a proponent of introducing something similar in the future.

I wonder if we introduce it as the field rendering, that we should also alter the passed in `data` object to only include the `dependencies` 🤷‍♂️

Reply:
Removed as part of this commit: https://github.com/WordPress/gutenberg/pull/65826/commits/3f73afb6da4a850088ada24bbbb21e85665935e5